daemon/api.go

```diff
@@ -416,7 +416,10 @@ func snapService(c *Command, r *http.Request) Response {
 
 	svcmap := make(map[string]*svcDesc, len(svcs))
 	for i := range svcs {
-		svcmap[svcs[i].Name] = &svcDesc{Spec: &svcs[i], Op: action}
+		if svcs[i].Daemon == "" {
+			continue
+		}
+		svcmap[svcs[i].Name] = &svcDesc{Spec: svcs[i], Op: action}
 	}
 
 	if svcName != "" && svcmap[svcName] == nil {
```

Comment:
appName, etc.. not sure about what else in the file needs updating, but worth a look.